### DISCUSSION TITLE
How to convert the BitmapImage to Image<Rgb24> in c# ?

### DISCUSSION THREAD

**Discussion by gitcheol** [SixLabors/ImageSharp#1605] [Q&A] [answered]
hi there. i'm newbie in ImageSharp.
In c#(WPF application), I`m using the ImageSharp to inference my deep learning model.
but i have some trouble.. 

This is the sample code(c#) 
https://www.onnxruntime.ai/docs/tutorials/resnet50_csharp.html

I want to change the input data part. 
`_Image<Rgb24> image = Image.Load<Rgb24>(imageFilePath, out IImageFormat format);_`

because my Image type is the BitmapImage Class (using System.Windows.Media.Imaging;)
`BitmapImage bitmap_image = new BitmapImage();`

So, How can i change from the BitmapImage(bitmap_image) to ImageSharp Image<Rgb24> ??? 

Have a wonderful day
Thank you!

**Comment by brianpopow** [comment]
@gitcheol can you be more specific what is not working and what you are doing? I was able to run the sample project from the tutorial you have posted using ImageSharp.

**Comment by antonfirsov** [comment, marked as answer]
@brianpopow I think the core of the question is about converting between WPF's `BitmapImage` and ImageSharp image, so then the sample code can be tried on it.

@gitcheol not familiar enough with WPF image sources, but I recommend to pull down and try the following project, it should contain working code for this:
https://github.com/jongleur1983/SharpImageSource

Also check the discussion under #531.

**Reply by gitcheol** [reply]
yes! you are right!! 
My question is the converting between WPF's BitmapImage and ImageSharp image.
Thank you so much!
Have a great weekent @antonfirsov @brianpopow
